modules/reindex/src/test/java/org/elasticsearch/index/reindex/ReindexFailoverIT.java

```diff
@@ -0,0 +1,129 @@
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class ReindexFailoverIT extends ReindexTestCase {
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        final HashSet<Class<? extends Plugin>> classes = new HashSet<>(super.nodePlugins());
+        classes.add(MockTransportService.TestPlugin.class);
+        return classes;
+    }
+
+    public void testReindexFailover() throws Throwable {
+        logger.info("--> start 4 nodes, 1 master, 3 data");
+
+        final Settings sharedSettings = Settings.builder()
+            .put("cluster.join.timeout", "10s")  // still long to induce failures but not too long so test won't time out
+            .build();
+
+        internalCluster().startMasterOnlyNodes(1, sharedSettings);
+        internalCluster().startDataOnlyNodes(3, sharedSettings);
+
+        logger.info("--> wait for all nodes to join the cluster");
+        ensureStableCluster(4);
+
+        ClusterStateResponse clusterState = client().admin().cluster().prepareState().setNodes(true).get();
+        Map<String, String> nodeIdToName = new HashMap<>();
+        for (ObjectObjectCursor<String, DiscoveryNode> node : clusterState.getState().nodes().getDataNodes()) {
+            nodeIdToName.put(node.key, node.value.getName());
+        }
+
+        // Create Index
+        Settings.Builder indexSettings = Settings.builder().put("index.number_of_shards", 1).put("index.number_of_replicas", 1);
+        client().admin().indices().prepareCreate("source").setSettings(indexSettings).get();
+        ensureGreen("source");
+
+        List<IndexRequestBuilder> docs = new ArrayList<>();
+        int docCount = between(1500, 5000);
+        for (int i = 0; i < docCount; i++) {
+            docs.add(client().prepareIndex("source", "_doc", Integer.toString(i)).setSource("foo", "a"));
+        }
+
+        indexRandom(true, docs);
+        assertHitCount(client().prepareSearch("source").setSize(0).get(), docCount);
+        ensureGreen("source");
+
+        // Copy all the docs
+        ReindexRequestBuilder copy = reindex().source("source").destination("dest").refresh(true);
+        StartReindexJobAction.Request request = new StartReindexJobAction.Request(copy.request(), false);
+
+        copy.source().setSize(100);
+        client().execute(StartReindexJobAction.INSTANCE, request).get();
+
+        ListTasksResponse listTasksResponse = client().admin().cluster().prepareListTasks().get();
+        Optional<TaskInfo> taskInfo = listTasksResponse.getTasks().stream().filter(t -> t.getAction().contains("reindex"))
+            .findFirst();
+        if (taskInfo.isPresent()) {
```

Comment:
Maybe we could throttle the reindex and then wait for a task to be present - so we are sure that the if is not bypassed.

Reply:
The task should be present now that we return the initial ephemeral task id in the response.